pkg/services/ngalert/notifier/channels/pushover.go

```diff
@@ -0,0 +1,246 @@
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+)
+
+const (
+	PUSHOVERENDPOINT = "https://api.pushover.net/1/messages.json"
+	BOUNDARY         = "abcd"
+)
+
+// PushoverNotifier is responsible for sending
+// alert notifications to Pushover
+type PushoverNotifier struct {
+	old_notifiers.NotifierBase
+	UserKey          string
+	APIToken         string
+	AlertingPriority int
+	OKPriority       int
+	Retry            int
+	Expire           int
+	Device           string
+	AlertingSound    string
+	OKSound          string
+	Upload           bool
+	Message          string
+	Title            string
+	tmpl             *template.Template
+	log              log.Logger
+}
+
+// NewSlackNotifier is the constructor for the Slack notifier
+func NewPushoverNotifier(model *NotificationChannelConfig, t *template.Template) (*PushoverNotifier, error) {
+	userKey := model.DecryptedValue("userKey", model.Settings.Get("userKey").MustString())
+	APIToken := model.DecryptedValue("apiToken", model.Settings.Get("apiToken").MustString())
+	device := model.Settings.Get("device").MustString()
+	alertingPriority, err := strconv.Atoi(model.Settings.Get("priority").MustString("0")) // default Normal
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert alerting priority to integer: %w", err)
+	}
+	okPriority, err := strconv.Atoi(model.Settings.Get("okPriority").MustString("0")) // default Normal
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert OK priority to integer: %w", err)
+	}
+	retry, _ := strconv.Atoi(model.Settings.Get("retry").MustString())
+	expire, _ := strconv.Atoi(model.Settings.Get("expire").MustString())
+	alertingSound := model.Settings.Get("sound").MustString()
+	okSound := model.Settings.Get("okSound").MustString()
+	uploadImage := model.Settings.Get("uploadImage").MustBool(true)
+
+	if userKey == "" {
+		return nil, alerting.ValidationError{Reason: "user key not found"}
+	}
+	if APIToken == "" {
+		return nil, alerting.ValidationError{Reason: "API token not found"}
+	}
+	return &PushoverNotifier{
+		NotifierBase: old_notifiers.NewNotifierBase(&models.AlertNotification{
+			Uid:                   model.UID,
+			Name:                  model.Name,
+			Type:                  model.Type,
+			DisableResolveMessage: model.DisableResolveMessage,
+			Settings:              model.Settings,
+			SecureSettings:        model.SecureSettings,
+		}),
+		UserKey:          userKey,
+		APIToken:         APIToken,
+		AlertingPriority: alertingPriority,
+		OKPriority:       okPriority,
+		Retry:            retry,
+		Expire:           expire,
+		Device:           device,
+		AlertingSound:    alertingSound,
+		OKSound:          okSound,
+		Upload:           uploadImage,
+		Message:          model.Settings.Get("message").MustString(`{{ template "default.message" .}}`),
+		Title:            model.Settings.Get("title").MustString(`{{ template "default.title" . }}`),
```

Comment:
We are not using this `Title` anywhere. So we can remove this variable.
Alternatively, add this as configurable and use this variable for title below.